hazelcast/src/main/java/com/hazelcast/topic/impl/TotalOrderedTopicProxy.java

```diff
@@ -47,4 +65,41 @@ public void publish(@Nonnull  E message) {
         InternalCompletableFuture f = invokeOnPartition(operation);
         f.joinInternal();
     }
+
+    @Override
+    public CompletionStage<Void> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
+        Operation op = new PublishOperation(getName(), toData(message))
+                .setPartitionId(partitionId);
+        return newDelegatingFuture(serializationService, publishInternalAsync(op));
+    }
+
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+        Operation op = new PublishAllOperation(getName(), toDataArray(messages));
```

Comment:
Since `PublishAllOperation` is a new operation and since this member might be running in a cluster with older members, we need to introduce the cluster version check:
```java
        Version clusterVersion = getNodeEngine().getClusterService().getClusterVersion();
        // RU_COMPAT_4_0
        if (!clusterVersion.isGreaterOrEqual(Versions.V4_1)){
            throw new UnsupportedOperationException("Publish all is not available on cluster version " + clusterVersion
                    + ". Please upgrade the cluster version to " + Versions.V4_1);
        }
```

Same in other methods using this operation.

Reply:
Got it - changed it.